### ISSUE TITLE
Can't search for add-on \ add to collection

### ISSUE THREAD

**Issue by irvinm** [mozilla/addons-server#6891]
### Describe the problem and steps to reproduce it:
Search for "View Image in New Tab" ([https://addons.mozilla.org/en-US/firefox/addon/new-tab-image_exe-boss/?src=search](url)) either from main add-on site or trying to add it to a collection.

### What happened?
Was not able to find the add-on or add to a collection.

### What did you expect to happen?
To find it.

### Anything else we should know?
Results from search:

![2017-11-12_8-34-57](https://user-images.githubusercontent.com/979729/32699502-e2554548-c784-11e7-844a-28481c921553.png)

I know there is\are other issue(s) regarding improving search results (including exact matches).  Is there any way to add an add-on to a collection without the search working properly?

**Comment by EnTeQuAk** [comment]
Thanks for the example! I'm working on https://github.com/mozilla/addons-server/issues/2661 so more examples to fix this problem once and for all.

Generally your problem will be fixed once https://github.com/mozilla/addons-server/issues/6837 got fixed which is high on the list right now.

**Comment by EnTeQuAk** [comment]
https://github.com/mozilla/addons-server/pull/7303 has landed on master and will be deployed on Thursday and should fix this problem. I'm closing this issue but please feel free to re-file / re-open any time if things aren't working properly.

**Comment by AlexandraMoga** [comment]
Verified on AMO-dev, FF58 (Win10)
1.  For search results in AMO please see my comments from [#6837](https://github.com/mozilla/addons-server/issues/6837#event-1447867918)
2.  Regarding search results for add-ons in the create collection form: 
- add-on with a name as an unbroken string (see [MyGiganticAddonName](https://addons-dev.allizom.org/en-US/firefox/addon/homepage-url/)) will not be found among search suggestions
- add-on containing 'stop words' (see [To do this and that](https://addons-dev.allizom.org/en-US/firefox/addon/permission-sync/) or [Search this on](https://addons-dev.allizom.org/en-US/firefox/addon/white-frost/)) will not be found among search suggestions
- exact matches are displayed but they don't always appear as first suggestions. I think this might still be an issue:

![image](https://user-images.githubusercontent.com/31961530/35566587-970ec404-05ca-11e8-8680-0c9e13e3e7c6.png)

**Comment by andreicristianpetcu** [comment]
[In my other bug](https://github.com/mozilla/addons-server/issues/7034) I wanted to add an addon to a collection. I really did not care about the UI I used to add it. I noticed I can add the addon to the collection from the addon's home page. Here is a print screen.

![screenshot from 2018-02-24 00-23-49](https://user-images.githubusercontent.com/922368/36619860-0d02c66a-18f9-11e8-8a74-eec143fd4b1c.png)

Not sure if this affects priority. I just wanted to share this :)
